catalog/dags/es/recreate_staging_index/recreate_full_staging_index.py

```diff
@@ -0,0 +1,95 @@
+from datetime import timedelta
+
+from airflow.decorators import task, task_group
+from airflow.exceptions import AirflowSensorTimeout
+from airflow.sensors.external_task import ExternalTaskSensor
+from airflow.utils.state import State
+
+from common import ingestion_server
+from common.sensors.utils import get_most_recent_dag_run
+from data_refresh.data_refresh_types import DATA_REFRESH_CONFIGS
+from database.staging_database_restore.constants import (
+    DAG_ID as STAGING_DB_RESTORE_DAG_ID,
+)
+
+
+DAG_ID = "recreate_full_staging_index"
+
+
+@task(retries=0)
+def prevent_concurrency_with_staging_database_restore(**context):
```

Comment:
I am going to make some sensor utils for doing these steps, since this will be useful in several places. I did not want to increase the complexity of this PR so I'll be doing that in a follow up.